### ISSUE TITLE
How close Unity Game (UnityView)

### ISSUE THREAD

**Issue by Vaibhaw-Redapple** [asmadsen/react-native-unity-view#9]
In React Native App at some page i have loaded UnityView. After exiting from that page, when user visits again that page then can we start the unity once again not resuming. As of now it keeps running until app is killed from memory.
Means on page exit can unity be destroyed ?

**Comment by rakirox** [comment]
Hi @Vaibhaw-Redapple , I don´t know if there is some one mataining this code (there are some interesting pr which I would like to merge). But this particular Issue is on my path I'll check it out and change the way it works or at least I'll make a method which destroys the unity instance. but it might take some time (my to do list is huge)

to @asmadsen  I would be happy to help you out mantaining this project. I'll be activly using it for a while I don't like the idea of having many forks with some small fixes. I think all effort should be in one place

**Comment by asmadsen** [comment]
Hi, @rakirox I'm no longer maintain this repo. However I'm happy to merge any pr with fixed as long as it doesn't break existing features

**Comment by rakirox** [comment]
@Vaibhaw-Redapple Just a heads up, i just solved this issue on android. In the next days I'll check it out for iOS. 

It will release must of unity memory usage and unload all scenes but it will keep in memory enough things to make a faster init the next time.

**Comment by paulfreund94** [comment]
@rakirox any chance you could share how you've gone about it / maybe open a PR?

**Comment by Apetrou** [comment]
@rakirox do you mind sharing how you achieved that?

**Comment by rakirox** [comment]
Yes! sorry been busy. tonight I'll draft a PR. I fixed only for android but haven't test it with iOS

**Comment by Apetrou** [comment]
Thanks so much @rakirox! Been trying a few things using native code but I keep killing the whole app!

**Comment by rakirox** [comment]
People, I just draft a PR which attack this problem for Android so you can use it before it's ready. 

I hope this weekend I'll have time to resolve al pending points.

cc: @Apetrou @paulfreund94 @Vaibhaw-Redapple 
[Draft Pull Request](https://github.com/asmadsen/react-native-unity-view/pull/16)

**Comment by nguyend3** [comment]
@rakirox Thank you for this! Any luck with iOS?

**Comment by zoix** [comment]
We solved this by just loading and unloading the main scene with `postMessageToUnityManager`:

```csharp
void MessageHandler(string message)
    {
        Debug.Log("onMessage: " + message);
        if (message == "start")
        {
            SceneManager.LoadScene("AR", LoadSceneMode.Additive);
        } else if (message == "stop")
        {
            SceneManager.UnloadSceneAsync("AR");
        }
    }
```

Not the most elegant solution but it works.

**Comment by gambit19** [comment]
Hi @asmadsen @rakirox Did you figure out a way to completely exit the Unity module on both Android and iOS? Thanks!

**Comment by DavidCotovanu** [comment]
> Hi @asmadsen @rakirox Did you figure out a way to completely exit the Unity module on both Android and iOS? Thanks!

Same idea with the MessageHandler from @agamadev answer, but you call `Application.Quit()` instead of `SceneManager.UnloadSceneAsync("AR")`. This will close the Unity app entirely.
